src/docfx/lib/schema/JsonSchemaValidator.cs

```diff
@@ -64,13 +66,15 @@ private static bool ValidateType(JsonSchema schema, JToken token, List<Error> er
             return true;
         }
 
-        private static void ValidateScalar(JsonSchema schema, List<Error> errors, JValue scalar)
+        private void ValidateScalar(JsonSchema schema, JValue scalar, List<Error> errors)
         {
             if (schema.Enum != null && !schema.Enum.Contains(scalar))
             {
                 errors.Add(Errors.UndefinedValue(JsonUtility.GetSourceInfo(scalar), scalar, schema.Enum));
             }
 
+            ValidateDateFormat(schema, scalar, errors);
```

Comment:
can put this under the if check in line 74, then you can access `str` without using `ToString()`. Also consider extract `ValidateFormat` and `ValidateLength`

Reply:
Sure, good suggestion